google/cloud/datastore/transaction.py

```diff
@@ -304,6 +366,7 @@ def commit(self, retry=None, timeout=None):
             # Clear our own ID in case this gets accidentally reused.
             self._id = None
 
+    @_begin_if_not_began
```

Comment:
Put and delete are not operations that make grpc calls. I am not sure we want to begin the transaction calls to these functions.

Reply:
@bhshkh As discussed offline we will want to remove the call to begin transaction from the flowchart to match [the other diagram](https://docs.google.com/document/d/1iJqA82FD2jYidz6J4kloDlO7lQQ_jnJxDgTeGMa_0wA/edit?resourcekey=0-X7Sl0mXsOd7WQIrk8bHDRg&tab=t.0#bookmark=id.hetxait3zwlu) right?